Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -106,7 +110,48 @@ buildRideListItem QRide.RideItem {..} = do
       }
 
 ---------------------------------------------------------------------------------------------------
+ticketRideList :: ShortId DM.Merchant -> Maybe (ShortId Common.Ride) -> Maybe Text -> Maybe Text -> Maybe Text -> Flow Common.TicketRideListRes
+ticketRideList merchantShortId mbRideShortId countryCode mbPhoneNumber _ = do
+  merchant <- findMerchantByShortId merchantShortId
+  let totalRides = 5
+  let mbShortId = coerce @(ShortId Common.Ride) @(ShortId DRide.Ride) <$> mbRideShortId
+  case mbPhoneNumber of
+    Just number -> do
+      no <- getDbHash number
+      let code = fromMaybe "+91" countryCode
+      person <- runInReplica $ QPerson.findByMobileNumberAndMerchant code no merchant.id >>= fromMaybeM (PersonWithPhoneNotFound number)
+      allRides <- QRide.findAllRidesByDriverId person.id
+      let sortedRides = sortBy (compare `on` (.createdAt)) allRides
+      let lastNRides = drop (length sortedRides - totalRides) sortedRides
+      ridesDetail <- mapM (\ride -> rideInfo merchantShortId (cast ride.id)) lastNRides
+      let rdList = zipWith (curry (makeRequiredRideDetail person.id)) lastNRides ridesDetail --map makeRequiredRideDetail (zip lastNRides ridesDetail)
+      return Common.TicketRideListRes {rides = rdList}
+    Nothing -> case mbShortId of
+      Just shortId -> do
+        ride <- QRide.findRideByRideShortId shortId >>= fromMaybeM (InvalidRequest "Ride ShortId Not Found")
+        rideDetail <- rideInfo merchantShortId (cast ride.id)
+        let ticketRideDetail = makeRequiredRideDetail ride.driverId (ride, rideDetail)
+        return Common.TicketRideListRes {rides = [ticketRideDetail]}
+      Nothing -> throwError $ InvalidRequest "Invalid Request"
+  where
+    makeRequiredRideDetail personId (ride, detail) =
+      Common.RideInfo
+        { rideShortId = mbRideShortId,
```

Comment:
You call this function five times for five last rides and everywhere use the same `mbRideShortId`